src/lib/index.d.ts

```diff
@@ -46,6 +47,20 @@ export type WebsocketAdapterConfig = {
   }
 }
 
+export type HttpAdapterConfig = {
+  server: {
+    httpServer?: any,
+    adapter?: HttpServerType | typeof GleeAdapter
+    port?: number
+  },
+  client?: {
+    auth?: {
```

Comment:
auth could be a function as well, I would suggest you take inspiration other adapter configs.